src/internal/superjson/superjson.ts

```diff
@@ -256,44 +257,74 @@ export class SuperJson {
     return this.normalizedCache;
   }
 
-  configHash(): string {
-    // <profile>:<version/path>,<provider>:<path>,<provider>:<path>
-    const profileValues: string[] = [];
-    for (const [profile, info] of Object.entries(this.normalized.profiles)) {
-      let path;
-      if ('version' in info) {
-        path = info.version;
-      } else {
-        path = info.file;
+  get anonymized(): AnonymizedSuperJsonDocument {
+    const profiles: AnonymizedSuperJsonDocument['profiles'] = {};
+    for (const [profile, profileEntry] of Object.entries(
+      this.normalized.profiles
+    )) {
+      const providers: typeof profiles[string]['providers'] = [];
+      for (const [provider, providerEntry] of Object.entries(
+        profileEntry.providers
+      )) {
+        const anonymizedProvider: typeof providers[number] = {
+          provider,
+          priority: profileEntry.priority.findIndex(
+            providerName => provider === providerName
+          ),
+        };
+        if ('file' in providerEntry) {
+          anonymizedProvider.version = 'file';
+        } else if ('mapRevision' in providerEntry) {
+          anonymizedProvider.version = providerEntry.mapRevision;
+          if (providerEntry.mapVariant !== undefined) {
+            anonymizedProvider.version += `-${providerEntry.mapVariant}`;
+          }
+        }
+
+        providers.push(anonymizedProvider);
       }
+      profiles[profile] = {
+        version: 'version' in profileEntry ? profileEntry.version : 'file',
+        providers,
+      };
+    }
 
-      const providers: string[] = Object.entries(info.providers).map(
-        ([provider, info]): string => {
-          if ('file' in info) {
-            return `${provider}:${info.file}`;
-          } else {
-            return `${provider}:${info.mapVariant ?? ''}-${
-              info.mapRevision ?? ''
-            }`;
-          }
+    return {
+      profiles,
+      providers: Object.keys(this.normalized.providers),
```

Comment:
Can normalized providers contain secrets - loaded from env or directly stored in super.json

Reply:
no, this is just keys == names